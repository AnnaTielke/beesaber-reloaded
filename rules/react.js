module.exports = {
  plugins: ['react'],

  extends: ['plugin:react/recommended'],

  settings: {
    react: {
      version: 'detect',
    },
  },

  rules: {
    'react/button-has-type': 'error',
    'react/jsx-boolean-value': 'error',
    'react/jsx-fragments': 'error',
    'react/jsx-handler-names': 'error',
    'react/jsx-no-bind': ['error', { allowArrowFunctions: true }],
    'react/jsx-pascal-case': 'error',
    'react/no-access-state-in-setstate': 'error',
    'react/no-array-index-key': 'error',
    'react/no-danger': 'error',
    'react/no-did-mount-set-state': 'error',
    'react/no-did-update-set-state': 'error',
    'react/no-multi-comp': 'error',
    'react/no-this-in-sfc': 'error',
    'react/no-typos': 'error',
    'react/no-unsafe': 'error',
    'react/no-unused-prop-types': 'error',
    'react/no-unused-state': 'error',
    'react/no-will-update-set-state': 'error',
    'react/prefer-es6-class': 'error',
    'react/prefer-stateless-function': 'error',
    'react/style-prop-object': 'error',
    'react/void-dom-elements-no-children': 'error',
  },
};
