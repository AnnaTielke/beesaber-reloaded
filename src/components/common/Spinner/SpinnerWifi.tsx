import styled from 'styled-components';

import tokens from '@/tokens/index';

const Container = styled.div`
  @keyframes ldio-reyiqutdbqg {
    0% {
      opacity: 0;
    }
    20% {
      opacity: 1;
    }
    100% {
      opacity: 1;
    }
  }
  .ldio-reyiqutdbqg div {
    position: absolute;
    animation: ldio-reyiqutdbqg 1s linear infinite;
    box-sizing: border-box !important;
  }
  .ldio-reyiqutdbqg div:nth-child(1) {
    width: 44px;
    height: 44px;
    left: 34px;
    top: 128px;
    border-radius: 50%;
    background: ${tokens.color.red.main};
  }
  .ldio-reyiqutdbqg div:nth-child(2) {
    width: 66px;
    height: 66px;
    left: 56px;
    top: 84px;
    border-radius: 0 66px 0 0;
    border: 20px solid ${tokens.color.red.shadow};
    background: none;
    animation-delay: 0.1s;
    border-bottom: 0;
    border-left: 0;
  }
  .ldio-reyiqutdbqg div:nth-child(3) {
    width: 110px;
    height: 110px;
    left: 56px;
    top: 40px;
    border-radius: 0 110px 0 0;
    border: 20px solid ${tokens.color.red.light};
    background: none;
    animation-delay: 0.2s;
    border-bottom: 0;
    border-left: 0;
  }
  .loadingio-spinner-radio-5b7w2tupa5t {
    width: 200px;
    height: 200px;
    display: inline-block;
    overflow: hidden;
    background: ${tokens.color.page.bgColor};
  }
  .ldio-reyiqutdbqg {
    width: 100%;
    height: 100%;
    position: relative;
    transform: translateZ(0) scale(1);
    backface-visibility: hidden;
    transform-origin: 0 0; /* see note above */
  }
  .ldio-reyiqutdbqg div {
    box-sizing: content-box;
  }
`;

function SpinnerWifi(): JSX.Element {
  return (
    <Container>
      <div className="loadingio-spinner-radio-5b7w2tupa5t">
        <div className="ldio-reyiqutdbqg">
          <div />
          <div />
          <div />
        </div>
      </div>
    </Container>
  );
}

export default SpinnerWifi;
