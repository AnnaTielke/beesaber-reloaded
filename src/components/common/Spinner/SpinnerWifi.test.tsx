import { render } from '@testing-library/react';

import SpinnerWifi from './SpinnerWifi';

describe('SpinnerWifi', () => {
  it('should match the snapshot', () => {
    const { container } = render(<SpinnerWifi />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
