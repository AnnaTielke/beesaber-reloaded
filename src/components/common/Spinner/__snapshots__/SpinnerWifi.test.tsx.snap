// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpinnerWifi should match the snapshot 1`] = `
.c0 .ldio-reyiqutdbqg div {
  position: absolute;
  -webkit-animation: ldio-reyiqutdbqg 1s linear infinite;
  animation: ldio-reyiqutdbqg 1s linear infinite;
  box-sizing: border-box !important;
}

.c0 .ldio-reyiqutdbqg div:nth-child(1) {
  width: 44px;
  height: 44px;
  left: 34px;
  top: 128px;
  border-radius: 50%;
  background: #ff0025;
}

.c0 .ldio-reyiqutdbqg div:nth-child(2) {
  width: 66px;
  height: 66px;
  left: 56px;
  top: 84px;
  border-radius: 0 66px 0 0;
  border: 20px solid #ff8d9e;
  background: none;
  -webkit-animation-delay: 0.1s;
  animation-delay: 0.1s;
  border-bottom: 0;
  border-left: 0;
}

.c0 .ldio-reyiqutdbqg div:nth-child(3) {
  width: 110px;
  height: 110px;
  left: 56px;
  top: 40px;
  border-radius: 0 110px 0 0;
  border: 20px solid #fff0f0;
  background: none;
  -webkit-animation-delay: 0.2s;
  animation-delay: 0.2s;
  border-bottom: 0;
  border-left: 0;
}

.c0 .loadingio-spinner-radio-5b7w2tupa5t {
  width: 200px;
  height: 200px;
  display: inline-block;
  overflow: hidden;
  background: #00111f;
}

.c0 .ldio-reyiqutdbqg {
  width: 100%;
  height: 100%;
  position: relative;
  -webkit-transform: translateZ(0) scale(1);
  -ms-transform: translateZ(0) scale(1);
  transform: translateZ(0) scale(1);
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
}

.c0 .ldio-reyiqutdbqg div {
  box-sizing: content-box;
}

<div
  class="c0"
>
  <div
    class="loadingio-spinner-radio-5b7w2tupa5t"
  >
    <div
      class="ldio-reyiqutdbqg"
    >
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;
